### PR TITLE
Datepicker should disappear. Create type-ahead for selecting city in Pla...

### PR DESCRIPTION
**Pull request by andrew-demidenko**
Datepicker should disappear. Create type-ahead for selecting city in Play create modal window. Remove input from goToDate.

### CODE REVIEW

File: client/static/app/modules/calendar/templates/calendar.directive.tpl.html [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-                       ng-model="dgCalendarIns.scope.date"
-                       view="month"
-                       max-view="year"
-                       min-view="month"
-                       format='MM/yyyy'>
-                <span class="input-group-addon">
+        <div class="bg-calendar__date-picker--wrapper" ng-hide="dgCalendarIns.editDateToggle">
+            <div ng-class="{open: !dgCalendarIns.editDateToggle}">
+                <ul class="dropdown-menu" role="menu">
+                    <div date-picker="dgCalendarIns.scope.date" view="month" max-view="year" min-view="month"></div>
```
Comment (artyomtrityak) [inline client/static/app/modules/calendar/templates/calendar.directive.tpl.html]: just `dgCalendarIns.date`

File: client/static/app/modules/city-picker/city-picker.directive.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ define(function(require) {
   'use strict';
 
   var cityPickerTemplate = require('text!./city-picker.tpl.html'),
+      cityPickerTemplate_1 = require('text!./city-picker-attribute-type.tpl.html'),
```
Comment (artyomtrityak) [inline client/static/app/modules/city-picker/city-picker.directive.js]: create another name

File: client/static/app/modules/play/controllers/play-new.controller.js [hunk 3]
```diff
@@ -1,12 +1,14 @@
 define(function(require) {
   'use strict';
-  
+
+  $ = require('jquery');
```
Comment (artyomtrityak) [inline client/static/app/modules/play/controllers/play-new.controller.js]: no, no $ in controller

File: client/static/app/modules/play/controllers/play-new.controller.js [hunk 4]
```diff
@@ -17,6 +19,12 @@ define(function(require) {
       isLoading: false
     };
 
+    $scope.$watch(function(){
+      return vm.playData.when;
+    }, function() {
+      $('input[name="game_date"]').blur();
```
Comment (artyomtrityak) [inline client/static/app/modules/play/controllers/play-new.controller.js]: don't do this way, this is not angular way

File: client/static/app/modules/play/templates/play-new.tpl.html [hunk 5]
```diff
@@ -95,6 +96,7 @@ <h3 class="m-t-none m-b">New Play</h3>
                         date-time
                         name="game_date"
                         ng-model="playNewIns.playData.when"
+                        ng-model-options="{ updateOn: 'blur'}"
```
Comment (artyomtrityak) [inline client/static/app/modules/play/templates/play-new.tpl.html]: why did you add it?

**Comment by artyomtrityak** [comment]
make changes and ping me for next round of review

File: client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<div class="dg-type-ahead">
+    <ui-select ng-model="dgCityPickerIns.city.selected"
+               theme="bootstrap"
+               reset-search-input="false"
+               on-select="dgCityPickerIns.onCitySelect($item, $model)"
+            >
+
+        <ui-select-match placeholder="For example: Odesa">
```
Comment (vadikace) [inline client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html]: Odessa ;)

File: client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+            {{$select.selected.name}}
+        </ui-select-match>
+        <ui-select-choices repeat="city in dgCityPickerIns.cities | filter: $select.search"
+                           refresh="dgCityPickerIns.getCities()"
+                           refresh-delay="0">
+            <div ng-bind-html="dgCityPickerIns.city.name | highlight: $select.search"></div>
```
Comment (vadikace) [inline client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html]: Instead of dgCityPickerIns.city.name use city.name and remove next line.

File: client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html (hunk 6 above)
Comment (artyomtrityak) [inline client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html]: where is your pull man?

File: client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<div class="dg-type-ahead">
```
Comment (vadikace) [inline client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html]: It's strange to create new template if you just need to change div container className. Should try another approach.

File: client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html (hunk 6 above)
Comment (vadikace) [inline client/static/app/modules/city-picker/city-picker-attribute-type.tpl.html]: on the way

File: client/static/app/modules/play/controllers/play-new.controller.js [hunk 9]
```diff
@@ -1,12 +1,12 @@
 define(function(require) {
   'use strict';
-  
+
   PlayNewController.$inject = [
-    '$rootScope', '$modalInstance', 'dgPlayService', 'startDate'
+    '$rootScope', '$scope', '$modalInstance', 'dgPlayService', 'startDate'
```
Comment (artyomtrityak) [inline client/static/app/modules/play/controllers/play-new.controller.js]: why do you need scope here?

**Comment by artyomtrityak** [comment]
remove scope and merge it, good job

File: client/static/app/modules/play/controllers/play-new.controller.js (hunk 9 above)
Comment (andrew-demidenko) [inline client/static/app/modules/play/controllers/play-new.controller.js]: When doing forget to remove
